clients/fides-js/src/components/tcf/TcfOverlay.tsx

```diff
@@ -87,12 +91,51 @@ export interface UpdateEnabledIds {
   modelType: keyof EnabledIds;
 }
 
+type Category =
+  | "purpose_consent"
+  | "purpose_legitimate_interests"
+  | "special_feature"
+  | "vendor_consent"
+  | "vendor_legitimate_interests"
+  | "system_consent"
+  | "system_legitimate_interests";
+type NoticeSubMap = {
+  [key: string]: string | number;
+};
+
+type NoticeMap = Record<Category, NoticeSubMap>;
+
+const noticeMap = (servedNotices: LastServedConsentSchema[]): NoticeMap => {
+  const map: NoticeMap = {
+    purpose_consent: {},
+    purpose_legitimate_interests: {},
+    special_feature: {},
+    vendor_consent: {},
+    vendor_legitimate_interests: {},
+    system_consent: {},
+    system_legitimate_interests: {},
+  };
+
+  servedNotices.forEach((notice) => {
+    (Object.keys(map) as Category[]).forEach((key) => {
+      const value = notice[key as keyof LastServedConsentSchema];
```

Comment:
it _might_ help TS to do something like this https://github.com/ethyca/fides/blob/main/clients/fides-js/src/lib/tcf/types.ts/#L177-L186 for `Category` and `LastServedConsentSchema`. that said, it might not get you as far as you want, sometimes TS is picky about this kind of thing. but it _might_ save you an `as` here or there

Reply:
Let's discuss this next week, I couldn't figure out a way to remove the `as keyof` without adding more code than we currently have for the related type definitions.

Reply:
okay no worries—I figured TS might get picky about it. let's leave it as-is then!